src/python/pants/option/custom_types.py

```diff
@@ -308,9 +308,19 @@ def __repr__(self):
     return '{} {}'.format(self.action, self.val)
 
 
-class GlobExpansionConjunction(enum('conjunction', ['any_match', 'all_match'])):
+class GlobExpansionConjunction(enum(['any_match', 'all_match'])):
+  """Describe whether to require that only some or all glob strings match in a target's sources.
 
-  # NB: The `default_value` is automatically the first element of the value list, but can be
-  # overridden or made more explicit in the body of the class. We want to be explicit here about
-  # which behavior we have when merging globs, as that can affect performance.
-  default_value = 'any_match'
+  NB: this object is interpreted from within Snapshot::lift_path_globs() -- that method will need to
+  be aware of any changes to this object's definition.
+  """
+
+  @classmethod
+  def create(cls, value=None):
+    # TODO: add testing for this value!
+    value = value or 'any_match'
+    # If passed an instance of the object, just return it. This makes copying glob expansion
```

Comment:
Sorry, I'm still not sure when this is useful... Surely if you already have a setting, you can just use it, and don't need to call `create` at all?

Reply:
There was at least one code path which may contain either an enum instance or a string for both of these types. I will swing back around to this and re-evaluate. I agree that it is not a good method to have and will try to remove it.

Reply:
Was able to remove these!